### PR TITLE
Flights list redesigned (closer to mock but still not final)

### PR DESCRIPTION
**Pull request by wojciech-kumon**

### CODE REVIEW

File: src/components/footer/Footer.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import FlexBar from '../common/FlexBar';
 import {appConfig} from '../../config/appConfig';
 import {GreyNewTabLink} from '../common/styledComponents';
 
-class Footer extends React.PureComponent {
+class Footer extends React.PureComponent<{}> {
```
Comment (patrykbuzowiczsabre) [inline src/components/footer/Footer.tsx]: why was this change needed?

File: src/components/flights/FlightsList.tsx [hunk 2]
```diff
@@ -1,85 +1,81 @@
...
 
 interface FlightsListProps {
+  formatMessage: FormatMessage;
   pricedItineraries: PricedItineraryType[];
   proposalCreation: ProposalCreationState;
 
   createProposal(options: CreateProposalOptions): void;
 }
 
+const buildItinsData = (pricedItineraries: PricedItineraryType[]) => {
```
Comment (patrykbuzowiczsabre) [inline src/components/flights/FlightsList.tsx]: we might want to extract these funcs to a separate file

File: src/components/footer/Footer.tsx (hunk 1 above)
Comment (wojciech-kumon) [reply src/components/footer/Footer.tsx]: React.PureComponent is like React.PureComponent<any> and allows passing any props (it's like java 1.4 without generics - I'm not sure how does it work in C#, so I can't give an example)

File: src/components/flights/FlightsList.tsx (hunk 2 above)
Comment (wojciech-kumon) [reply src/components/flights/FlightsList.tsx]: Yes, but this function will be refactored/changed (more fields are needed)

Comment (patrykbuzowiczsabre) [reply src/components/flights/FlightsList.tsx]: sure

File: src/components/footer/Footer.tsx (hunk 1 above)
Comment (patrykbuzowiczsabre) [reply src/components/footer/Footer.tsx]: I can see
image
and
image
don't you have exactly the same?
Also I can see that the non-generic version is actually the generic one with default type args
image

Comment (wojciech-kumon) [reply src/components/footer/Footer.tsx]: ok, you're right, there default `P = {}` in typings
